### DISCUSSION TITLE
Spoofing HTTPS in dev on ios

### DISCUSSION THREAD

**Discussion by jayenashar** [ionic-team/capacitor#3464] [Q&A]
I am using HTTPS in production as described here: https://github.com/ionic-team/capacitor/discussions/1478#discussioncomment-24247
basically by setting `server.url` to https://$hostname/

I noticed that ios in development uses the capacitor protocol/scheme as described here: https://github.com/ionic-team/capacitor/issues/758#issuecomment-444851005

Can I test my HTTPS-dependent stuff in development on ios?  Like, for geolocation, I polyfill `navigator.geolocation` to use capacitor's geolocation (if `isPlatform(ios) && !isPlatform(mobileweb)`, but them i'm not really testing how it will be in production.

**Comment by jcesarmobile** [comment]
Capacitor always use capacitor protocol, in development and in production.
If you configure it to point to a https server it will also use it in development and in production.
If you want to test different things in development and in productions just use different https urls.

But pointing to a web server in production apps is not really recommended and can lead to app rejections.

**Reply by jayenashar** [reply]
> Capacitor always use capacitor protocol, in development and in production.
> If you configure it [capacitor] to point to a https server it [capacitor] will also use it [?] in development and in production.

Sorry I'm a bit confused here.  What is the last `it` here?  Is it the capacitor protocol, or https server?

> If you want to test different things in development and in productions just use different https urls.

I want to test the same thing.  I want development to be as close to production as possible.  I'm trying to figure out how to test features in development that depend on https in production.

> But pointing to a web server in production apps is not really recommended and can lead to app rejections.

Understood.  Thanks.  I've got cordova apps working this way without rejection so hopefully this capacitor app will be the same. :crossed_fingers:

**Reply by jcesarmobile** [reply]
If you configure capacitor to use the server url, it will always load the server url instead of the local assets, won’t matter if it’s a development or production build. So if you want development to be as close to production as possible, you’ll have to also host it on a server and change the server url to point to the development url.

**Reply by pakastin** [reply]
Will geolocation work in production? In development it's saying:
```
[blocked] Access to geolocation was blocked over insecure connection to capacitor://localhost.
```

**Reply by jcesarmobile** [reply]
Web geolocation won’t work, Apple is supposed to fix it, but no timeline.
But you can use geolocation plugin.

**Reply by pakastin** [reply]
Plugin returns the same error as well

**Reply by jcesarmobile** [reply]
Then you are using the plugin in a wrong way that uses the web implementation instead of the native implementation. Most likely you are using geolocation object directly instead of picking it from Plugins object as the docs explain.

**Reply by pakastin** [reply]
I do use the plugin, but somehow that calls window.navigator.geolocation
<img width="1068" alt="Näyttökuva 2021-3-17 kello 15 16 05" src="https://user-images.githubusercontent.com/1475902/111473431-b3874280-8733-11eb-8fed-a9149b74e1c2.png">

**Reply by jcesarmobile** [reply]
read my previous message again

**Reply by pakastin** [reply]
See the source tree left, it's in capacitor.js code

**Reply by jcesarmobile** [reply]
If you don't use the plugins in the way the docs explain, then the plugins use the web version instead of the native version, and the web version uses `navigator.geolocation`.
Your screenshot only proves that you are using the plugin in the wrong way.

**Reply by pakastin** [reply]
It seems adding app bound domain (to get service worker working) breaks it. I removed app bound domain and now it works 🤔

**Comment by jayenashar** [comment]
Ok, understood.  Thanks.  I'm using a create-react-app-based framework, so I should be able to start the dev server with HTTPS and then point capacitor to it.  Thanks.
